packages/playwright-core/src/server/chromium/crServiceWorker.ts

```diff
@@ -116,6 +116,6 @@ export class CRServiceWorker extends Worker {
   }
 
   private _isNetworkInspectionEnabled(): boolean {
-    return this._browserContext._options.serviceWorkers === 'allow';
+    return !!process.env.PW_EXPERIMENTAL_SERVICE_WORKER_NETWORK_EVENTS && this._browserContext._options.serviceWorkers === 'allow';
```

Comment:
We should also not use a network manager on this session as constructing it will trigger a bunch of network events.